packages/gatsby/src/utils/api-browser-docs.js

```diff
@@ -17,6 +17,45 @@ exports.onClientEntry = true
  */
 exports.onInitialClientRender = true
 
+/**
+ * Called when changing location is started.
+ * @param {object} $0
+ * @param {object} $0.location A location object
+ * @param {object} $0.action The "action" that caused the route change
+ * @example
+ * exports.onPreRouteUpdate = ({ location }) => {
+ *   console.log("Gatsby started to change location", location.pathname)
+ * }
+ */
+exports.onPreRouteUpdate = true
+
+/**
+ * Called when changing location is longer than 1 second.
+ * @param {object} $0
+ * @param {object} $0.location A location object
+ * @param {object} $0.action The "action" that caused the route change
+ * @example
+ * exports.onRouteUpdateDelayed = () => {
+ *   console.log("We can show loading indicator now")
+ * }
+ */
+exports.onRouteUpdateDelayed = true
+
+/**
+ * Called when error is thrown during location change
+ *
+ * Most likely it is caused by navigating to path that doesn't exist
+ * @param {object} $0
+ * @param {object} $0.location A location object
+ * @param {object} $0.action The "action" that caused the route change
+ * @param {object} $0.error An error object
+ * @example
+ * exports.onRouteUpdateError = ({ location, error }) => {
```

Comment:
Should the examples include every param? e.g. adding `action` to this one `({ location, action, error })`

Reply:
I checked other examples and `onRouteUpdate` (that I didn't change at all) doesn't use all params ( https://github.com/pieh/gatsby/blob/5007dbffcfe9d2985f56997412362e20be728c66/packages/gatsby/src/utils/api-browser-docs.js#L65 ). But given that we only console log in those examples it's no problem to add all params there.

Reply:
oh ok, maybe leave this as-is if that's how it's done elsewhere.